pkg/term/windows/ansi_writer.go

```diff
@@ -0,0 +1,454 @@
+	if toCoord.Y < fromCoord.Y {
+		return nil
+	}
+
+	var err error
+
+	var coordStart = COORD{}
+	var coordEnd = COORD{}
+
+	xCurrent, yCurrent := fromCoord.X, fromCoord.Y
+	xEnd, yEnd := toCoord.X, toCoord.Y
+
+	// Clear any partial initial line
+	if xCurrent > 0 {
+		coordStart.X, coordStart.Y = xCurrent, yCurrent
+		coordEnd.X, coordEnd.Y = xEnd, yCurrent
+
+		err = aw.clearRect(attributes, coordStart, coordEnd)
+		if err != nil {
+			return err
+		}
+
+		xCurrent = 0
+		yCurrent += 1
+	}
+
+	// Clear intervening rectangular section
+	if yCurrent < yEnd {
+		coordStart.X, coordStart.Y = xCurrent, yCurrent
+		coordEnd.X, coordEnd.Y = xEnd, yEnd-1
+
+		err = aw.clearRect(attributes, coordStart, coordEnd)
+		if err != nil {
+			return err
+		}
+
+		xCurrent = 0
+		yCurrent = yEnd
+	}
+
+	// Clear remaining partial ending line
+	coordStart.X, coordStart.Y = xCurrent, yCurrent
+	coordEnd.X, coordEnd.Y = xEnd, yEnd
+
+	err = aw.clearRect(attributes, coordStart, coordEnd)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (aw *ansiWriter) clearRect(attributes WORD, fromCoord COORD, toCoord COORD) error {
+	region := SMALL_RECT{Top: fromCoord.Y, Left: fromCoord.X, Bottom: toCoord.Y, Right: toCoord.X}
+
+	width := toCoord.X - fromCoord.X + 1
+	height := toCoord.Y - fromCoord.Y + 1
+	size := uint32(width) * uint32(height)
+	buffer := make([]CHAR_INFO, size)
+
+	if size <= 0 {
+		return nil
+	}
+
+	char := CHAR_INFO{WCHAR(FILL_CHARACTER), attributes}
+	for i := 0; i < int(size); i++ {
+		buffer[i] = char
+	}
+
+	err := WriteConsoleOutput(aw.fd, buffer, COORD{X: width, Y: height}, COORD{X: 0, Y: 0}, &region)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// doAnsiCommand translates ANSI commands into Windows API calls.
+func (aw *ansiWriter) doAnsiCommand() (err error) {
+
+	info, err := GetConsoleScreenBufferInfo(aw.fd)
+	if err != nil {
+		return err
+	}
+
+	ac := newAnsiCommand(aw.command)
+	logrus.Debugf("[windows] doAnsiCommand: Cmd(%v)", ac)
+
+	switch ac.Command {
+	case "A", "B", "C", "D", "E", "F", "G":
+		// [incrementA -- Move up
+		// [incrementB -- Move down
+		// [incrementC -- Move right
+		// [incrementD -- Move left
+		// [incrementE -- Move to (0, y+increment)
+		// [incrementF -- Move to (0, y-increment)
+		// [columnG -- Move to column (1-based) in current line
+
+		value := ac.paramAsSHORT(0, 1)
+
+		if ac.Command == "A" || ac.Command == "D" || ac.Command == "F" {
+			value = -value
+		}
+
+		position := info.CursorPosition
+		if ac.Command == "C" || ac.Command == "D" {
+			position.X = addInRange(position.X, value, info.Window.Left, info.Window.Right)
+		} else if ac.Command == "G" {
+			position.X = addInRange(value, -1, info.Window.Left, info.Window.Right)
+		} else {
+			if ac.Command == "E" || ac.Command == "F" {
+				position.X = 0
+			}
+			position.Y = addInRange(position.Y, value, info.Window.Top, info.Window.Bottom)
+		}
+
+		err = aw.setCursorPosition(position, info.Size)
```

Comment:
Continuing discussion from https://github.com/docker/docker/pull/12602/files#r29278853 :

I suggest you send `&position` instead of `position`.